ui/library/src/context/ScrollContext.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+
+import { Nullable } from '../components/types/utils';
+import { logProviderWarning } from '../utils/provider';
+import ScrollDetector from '../utils/ScrollDirectionDetector';
+
+export enum ScrollDirection {
+  UP = 'UP',
+  DOWN = 'DOWN',
+}
```

Comment:
I think this should be moved to the `ScrollDirectionDetector`. It's kinda owning the direction stuff now.